R/database.R

```diff
@@ -2628,3 +2630,385 @@ get_gene_expression = function(metadata,
+      }
+    }
+  }
+
+  #last check
+  if(missing(this_sample_id) && missing(these_samples_metadata)){
+    stop("please provide either a sample ID (this_sample_id) or an already subset metadata (these_samples_metadata)...")
+  }
+
+  #get wildcards
+  tumour_sample_id = this_sample_id
+  unix_group = pull(these_samples_metadata, unix_group)
+  seq_type = pull(these_samples_metadata, seq_type)
+  genome_build = projection
+  pairing_status = pull(these_samples_metadata, pairing_status)
+
+  if(pairing_status == "matched"){
+    normal_sample_id = pull(these_samples_metadata, normal_sample_id)
+  }else{
+    normal_sample_id = config::get("unmatched_normal_ids")[[unix_group]][[seq_type]][[genome_build]]
+  }
+
+  if(from_flatfile){
+    if(!remote_session){
+      path_template = config::get("results_flatfiles")$sv_manta$template
+      path_template = paste0(config::get("project_base"), path_template)
+      bedpe_path = glue::glue(path_template)
+
+      #if file does not exist, look for it in the merge (already loaded and given to the all_sv parameter, should be in the genome projection your interested in).
+      if(!file.exists(bedpe_path)){
+        if(missing(all_sv_grch37) && missing(all_sv_hg38)){
+          stop("Considder use all_sv parameter for retreiving SV calls for missing samples...")
+        }else{
+          if(projection == "grch37"){
+            bedpe_dat = all_sv_grch37 %>%
+              dplyr::filter(normal_sample_id == this_sample_id) #look for this_sample_id in the normal_sample_id column.
+            if(!nrow(bedpe_dat==0)){ #if sample not available, look for it in the tumour_sample_id.
+              bedpe_dat = all_sv_grch37 %>%
+                dplyr::filter(tumour_sample_id == this_sample_id)
+            }
+          }
+          if(projection == "hg38"){
+            bedpe_dat = all_sv_hg38 %>%
+              dplyr::filter(normal_sample_id == this_sample_id) #look for this_sample_id in the normal_sample_id column.
+            if(!nrow(bedpe_dat==0)){ #if sample not available, look for it in the tumour_sample_id.
+              bedpe_dat = all_sv_hg38 %>%
+                dplyr::filter(tumour_sample_id == this_sample_id)
+            }
+          }
+        }
+
+        #if can't find the sample in the existing merge, look for individual files (in the opposite projection) and lift over to the desired projection.
+        if(!nrow(bedpe_dat==0)){
+          if(projection == "grch37"){
+            genome_build = "hg38"
+            bedpe_path = glue::glue(path_template)
+            genome_build = "grch37"
+            bedpe_dat = liftover_bedpe(bedpe_file = bedpe_path, target_build = "grch37")
+
+          }else if(projection == "hg38"){
+            genome_build = "grch37"
+            bedpe_path = glue::glue(path_template)
+            genome_build = "hg38"
+            bedpe_dat = liftover_bedpe(bedpe_file = bedpe_path, target_build = "hg38")
+          }
```

Comment:
As you implemented it, this function is going to be running liftover many times on the individual files rather than just once, right?

Reply:
Yes, I couldn't think of any other way to do this. I know ideally, we would want to use liftover_bedpe on all sv calls that would need to lifted (instead of running this on individual files). Maybe it would be all right since this step would only be used in "rare cases" when a sample is not available in the merge, or individual flatfile (in the desired projection)? If not, I would be happy to take suggestions on how this can be achieved.

Reply:
is this resolved?